tests/timeseries/test_quantities.py

```diff
@@ -42,6 +42,56 @@ class Fz2(
     """Frequency quantity with broad exponent unit map."""
 
 
+def test_zero() -> None:
+    """Test the zero value for quantity."""
+    assert Quantity(0.0) == Quantity.zero()
+    assert Quantity(0.0, exponent=100) == Quantity.zero()
+    assert Quantity.zero() is Quantity.zero()  # It is a "singleton"
+    assert Quantity.zero().base_value == 0.0
+
+    assert Power.from_watts(0.0) == Power.zero()
+    assert Power.from_kilowatts(0.0) == Power.zero()
+    assert isinstance(Power.zero(), Power)
+    assert Power.zero().as_watts() == 0.0
+    assert Power.zero().as_kilowatts() == 0.0
+    assert Power.zero() is Power.zero()  # It is a "singleton"
+
+    assert Current.from_amperes(0.0) == Current.zero()
+    assert Current.from_milliamperes(0.0) == Current.zero()
+    assert isinstance(Current.zero(), Current)
+    assert Current.zero().as_amperes() == 0.0
+    assert Current.zero().as_milliamperes() == 0.0
+    assert Current.zero() is Current.zero()  # It is a "singleton"
+
+    assert Voltage.from_volts(0.0) == Voltage.zero()
+    assert Voltage.from_kilovolts(0.0) == Voltage.zero()
+    assert isinstance(Voltage.zero(), Voltage)
+    assert Voltage.zero().as_volts() == 0.0
+    assert Voltage.zero().as_kilovolts() == 0.0
+    assert Voltage.zero() is Voltage.zero()  # It is a "singleton"
+
+    assert Energy.from_kilowatt_hours(0.0) == Energy.zero()
+    assert Energy.from_megawatt_hours(0.0) == Energy.zero()
+    assert isinstance(Energy.zero(), Energy)
+    assert Energy.zero().as_kilowatt_hours() == 0.0
+    assert Energy.zero().as_megawatt_hours() == 0.0
+    assert Energy.zero() is Energy.zero()  # It is a "singleton"
+
+    assert Frequency.from_hertz(0.0) == Frequency.zero()
+    assert Frequency.from_megahertz(0.0) == Frequency.zero()
+    assert isinstance(Frequency.zero(), Frequency)
+    assert Frequency.zero().as_hertz() == 0.0
+    assert Frequency.zero().as_megahertz() == 0.0
+    assert Frequency.zero() is Frequency.zero()  # It is a "singleton"
+
+    assert Percentage.from_percent(0.0) == Percentage.zero()
+    assert Percentage.from_fraction(0.0) == Percentage.zero()
+    assert isinstance(Percentage.zero(), Percentage)
+    assert Percentage.zero().as_percent() == 0.0
+    assert Percentage.zero().as_fraction() == 0.0
+    assert Percentage.zero() is Percentage.zero()  # It is a "singleton"
```

Comment:
```py
x = Percentage.zero()
x = x + Percentage.from_percent(1)
assert Percentage.zero() == Percentage.from_percent(0)
```

would this be assert fail?

Reply:
I think it wouldn't,  because `x + ...` generates a new object,  and that gets bound to `x`,  and the zero object's reference count would just get decremented.  It is not possible to get the deref behaviour of other languages in Python I think (at least easily).

Reply:
but true, with `x = x + ..` it doesn't fail

Reply:
:o I forgot we implemented +=, etc.  So now we have to decide whether to keep += or not cache zeros?

Reply:
You could do something terrible and check in `+=` if you are a singleton and reset the according cache

Reply:
You can't change the object returned by `Percentage.zero()` unless you do this: `Percentage._zero_cache[Percentage] = x`. Variables in Python are bound to objects, you are just rebinding `x` when doing `x = ...`, nothing more.

Reply:
@llucax we have `+=` now, which mutate the objects.  that's what @Marenz is showing in his example above.

Reply:
OK, I didn't see the `+=`, that is a problem indeed because it is mutating the object, so yeah, if objects are not immutable it is an issue. I guess we then need to drop the singleton idea, good catch. I guess it is not **that** bad.

Or, drop mutable method. I'm not sure really, there are very nice properties of objects being immutable (like being able to use them as defaults in function arguments). Now I wonder how built-in types work, because they are immutable but `+=` also work on them.

Reply:
I guess you can just return a new object in `__iadd__`, etc. Should we do that?

Reply:
It seems like `__i*__` are the only mutating methods. I think for number we really want them to be immutable.

Reply:
I just tried removing all `__i*__` methods and then `x += ...` with `x = Quantity.zero()` works. I really think we should do that. I think users would expect to be able to use defaults in function arguments, I think `x += ...` should be rare enough to trade a bit of worse performance for safety.

Reply:
you mean `x += ...` works without `__iadd__`,  and acts as alias for `x = x + ...`?  If that's the case, we can remove that.  Maybe in this PR?

Reply:
Done.